src/components/_shared/TabBar/TabBar.stories.tsx

```diff
@@ -10,45 +10,50 @@ export default {
 }
 
 export const Basic = () => {
-  const tabs = array('tabs', ['one', 'two', 'three'])
+  const [selectedTab, setSelectedTab] = useState('one')
+  const tabs = ['one', 'two', 'three']
 
   return (
     <TabBar
-      id='tab-bar-id'
-      class='custom-class-name'
+      className='custom-class-name'
       title='select something'
-      selectedTab={text('selectedTab', 'one')}
+      selectedTab={selectedTab}
       tabs={tabs}
-      onChange={action('changed')}
+      onChange={setSelectedTab}
     />
   )
 }
 
 export const WithCustomColor = () => {
+  const [selectedTabIndex, setSelectedTabIndex] = useState(0)
   const tabs = [
     { color: 'red', display: 'Option number one' },
     { color: 'blue', display: 'Option number two' },
     { color: 'green', display: 'Option number three' },
   ]
-
-  const selectedTabIndex = number('selectedTabIndex', 0)
   const selectedTab = tabs[selectedTabIndex]
 
   return (
     <TabBar
-      id='tab-bar-id'
-      class='custom-class-name'
+      className='custom-class-name'
       title='select something'
       selectedTab={selectedTab}
       tabs={tabs}
       format={(tab) => tab.display}
-      color={(tab) => (tab === selectedTab ? tab.color : null)}
-      onChange={action('changed')}
+      color={(tab) => (tab === selectedTab ? tab.color : colors.darkGrey)}
+      onChange={(tab) => {
+        // for some reason setState will make a copy or something so
+        // so selectedTab === tab won't work anymore inside the component
+        // we can't just compare tabs but have to use the index
+        const newIndex = tabs.findIndex((t) => t === tab)
+        setSelectedTabIndex(newIndex)
```

Comment:
I first misunderstood the issue: I thought you had to change the actual `TabBar` implementation so it worked with the storybook. The fact that you didn't means that we're probably using a good abstraction that's not leaking. :)